fem/quadinterpolator_det.cpp

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "quadinterpolator.hpp"
+#include "../general/forall.hpp"
+#include "../linalg/dtensor.hpp"
+#include "../fem/kernels.hpp"
+#include "../linalg/kernels.hpp"
+
+using namespace mfem;
+
+namespace mfem
+{
+
+template<int T_D1D = 0, int T_Q1D = 0, int MAX_D1D = 0, int MAX_Q1D = 0>
+static void Det2D(const int NE,
+                  const double *b,
+                  const double *g,
+                  const double *x,
+                  double *y,
+                  const int vdim = 1,
+                  const int d1d = 0,
+                  const int q1d = 0)
+{
+   constexpr int DIM = 2;
+   constexpr int NBZ = 1;
+
+   const int D1D = T_D1D ? T_D1D : d1d;
+   const int Q1D = T_Q1D ? T_Q1D : q1d;
+
+   const auto B = Reshape(b, Q1D, D1D);
+   const auto G = Reshape(g, Q1D, D1D);
+   const auto X = Reshape(x,  D1D, D1D, DIM, NE);
+   auto Y = Reshape(y, Q1D, Q1D, NE);
+
+   MFEM_FORALL_2D(e, NE, Q1D, Q1D, NBZ,
+   {
+      constexpr int NBZ = 1;
```

Comment:
Defined twice

Reply:
This is needed for the `AppVeyor` builds.

Reply:
I think you can use `static constexpr int NBZ = 1` and you don't need to define it two times like [this](https://github.com/mfem/mfem/blob/master/fem/restriction.cpp#L286).